crates/sequencer/src/sequencer.rs

```diff
@@ -238,7 +234,23 @@ where
             da_block.header(),
             &mut signed_batch,
         ) {
-            (Ok(()), batch_workspace) => {
+            (Ok(()), mut batch_workspace) => {
+                let system_tx_gas_usage = self
```

Comment:
put comment here that explains if there's going to be system txs somewhere else then the beginning of the block, then only thing wouldn't work